application/src/main/java/com/sanction/thunder/resources/VerificationResource.java

```diff
@@ -39,13 +46,100 @@
    * @param metrics The metrics object to set up meters with.
    */
   @Inject
-  public VerificationResource(PilotUsersDao usersDao, MetricRegistry metrics) {
+  public VerificationResource(PilotUsersDao usersDao,
+                              MetricRegistry metrics,
+                              EmailService emailService) {
     this.usersDao = usersDao;
+    this.emailService = emailService;
 
     // Set up metrics
-    this.postRequests = metrics.meter(MetricRegistry.name(
+    this.getRequests = metrics.meter(MetricRegistry.name(
         UserResource.class,
-        "post-requests"));
+        "get-requests"));
+  }
+
+  /**
+   * Validates a user account by sending an email with a unique token.
+   *
+   * @param key The basic authentication key necessary to access the resource.
+   * @param email The email to send a unique token to.
+   * @return A response status and message.
+   */
+  @POST
+  public Response verifyUser(@Auth Key key,
+                             @QueryParam("email") String email,
+                             @HeaderParam("password") String password) {
+    getRequests.mark();
+
+    if (email == null || email.isEmpty()) {
+      LOG.warn("Attempted user verification without an email.");
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity("Incorrect or missing email query parameter.").build();
+    }
+
+    if (password == null || password.isEmpty()) {
+      LOG.warn("Attempted to verify user {} without a password.", email);
+      return Response.status(Response.Status.BAD_REQUEST)
+          .entity("Incorrect or missing header credentials.").build();
+    }
+
+    LOG.info("Attempting to verify user {}", email);
+
+    // Get the existing PilotUser
+    PilotUser user;
+    try {
+      user = usersDao.findByEmail(email);
+    } catch (DatabaseException e) {
+      LOG.error("Error retrieving user {} in database. Caused by: {}", email, e.getErrorKind());
+      return e.getErrorKind().buildResponse(email);
+    }
+
+    // Generate the unique verification token
+    String token = generateVerificationToken();
+
+    // Update the user's verification token
+    PilotUser updatedUser = new PilotUser(
+        new Email(user.getEmail().getAddress(), false, token),
+        user.getPassword(),
+        user.getFacebookAccessToken(),
+        user.getTwitterAccessToken(),
+        user.getTwitterAccessSecret()
+    );
+
+    PilotUser result;
+    try {
+      result = usersDao.update(user.getEmail().getAddress(), updatedUser);
+    } catch (DatabaseException e) {
+      LOG.error("Error posting user {} to the database. Caused by {}",
+          user.getEmail(), e.getErrorKind());
+      return e.getErrorKind().buildResponse(user.getEmail().getAddress());
+    }
+
+    // Send the token URL to the users email
+    boolean emailResult = emailService.sendEmail(result.getEmail(),
+        "Account Verification",
+        new StringJoiner("\n")
+        .add("<h1> Welcome to Pilot! </h1>")
```

Comment:
I feel like these `.add()` should be indented by two spaces so that it's not right under the `new StringJoiner` line. Because these are continuation lines for that